### PR TITLE
Restructure the repo to be less Docker focused.

### PR DESCRIPTION
**Pull request by mattmoor**
This restructures the repository to be less docker-centric.  We may eventually want to move this repo to `@io_bazel_rules_containers`.

The general structure is going to be:
```
  # The bulk of the repo, defining format-agnostic concepts,
  # e.g. container_image, container_push, ...
  container/...

  # Docker-specific aliases (docker_build, docker_push, ...)
  docker/docker.bzl

  # OCI-specific aliases (oci_image, oci_push, ...)
  oci/oci.bzl

  # Shared language .bzl files
  lang/
  # Language-specific rules
  {lang}/

  # Test data
  testdata/...

  # Unspecified-format contributions
  contrib/...
```

Fixes: https://github.com/bazelbuild/rules_docker/issues/110

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,10 +29,10 @@ images.**  This means:
 `boot2docker` or `docker-machine` installed.
 * They do not require root access on your workstation.
 
-Also, unlike traditional Docker builds, the Docker images produced by
-`docker_build` are deterministic / reproducible.
+Also, unlike traditional container build (e.g. Dockerfile), the Docker images
```
Comment (jonjohnsonjr) [inline README.md]: s/build/builds/

File: README.md [hunk 2]
```diff
@@ -76,9 +87,9 @@ your local Docker client by running:
 `docker load -i bazel-genfiles/my/image/helloworld.tar`.  Building
 this target can be expensive for large images.
 
-These work with both `docker_build` and `docker_bundle`.  For
-`docker_build` the image name will be `bazel/my/image:helloworld`.
-For `docker_bundle`, it will apply the tags you have specified.
+These work with both `container_image` and `container_bundle`.  For
+`container_image` the image name will be `bazel/my/image:helloworld`.
```
Comment (jonjohnsonjr) [inline README.md]: Consistency: add comma after `container_image`

File: README.md [hunk 3]
```diff
@@ -1,16 +1,23 @@
-# Bazel Docker Rules
+# Bazel Container Image Rules
```
Comment (jonjohnsonjr) [inline README.md]: Just "Bazel Container Rules" ?

File: README.md [hunk 4]
```diff
@@ -359,11 +376,11 @@ use with `docker_build`'s `base` attribute.
   </tbody>
 </table>
 
-<a name="docker_push"></a>
-## docker_push
+<a name="container_push"></a>
```
Comment (jonjohnsonjr) [inline README.md]: Does this do anything?

**Comment by mattmoor** [comment]
@damienmg @dlorenc FYI.

I may also move the `{cc,java,py,go}_image` rules out of `contrib/`, make them top-level directories, and put together documentation for them.

File: README.md [hunk 5]
```diff
@@ -387,6 +404,14 @@ An executable rule that pushes a Docker image to a Docker registry on `bazel run
       </td>
     </tr>
     <tr>
+      <td><code>format</code></td>
+      <td>
+        <p><code>Kind, required</code></p>
+        <p>The format as which to publish the image.  Currently, this supports
```
Comment (jonjohnsonjr) [inline README.md]: "The format of the published image." or even
"The desired format of the published image."?

File: README.md (hunk 3 above)
Comment (mattmoor) [reply README.md]: This doesn't do stuff like Kubernetes, so I scoped it to images.

**Comment by damienmg** [comment]
This is a big breaking change, Maybe still offer the ability to load the image with the old name. E.g load("....bzl", docker_build ="container_image") in docker.bzl?

**Comment by mattmoor** [comment]
@damienmg That's exactly what I have in place :)

I also allow folks to load `oci/oci.bzl` and `oci_*`, but these are all aliases (except `oci_push`, which is an alias w/ one default kwarg).

**Comment by mattmoor** [comment]
@damienmg Just to be completely explicit, there are a handful of things that are going to break with this change, which I've convinced myself is an acceptable trade-off:
```
# Contrib loads will fail because all of the files moved, this should be as simple
# as dropping docker/.  EXCEPT for the foo_image rules, which I want to graduate
# into first-class rules here
load("//docker/contrib:*", "*")

# If I circumvented the docker.bzl file, this will break.  Instead use docker.bzl.
load("//docker:build.bzl", "...")
```

I'm not aware of anything else that will break.

**Comment by mattmoor** [comment]
@damienmg Yes, I thought about that, we'll want to think about how we extract that, what we use for OCI, etc, etc.

I've pushed the language `foo_image` changes, so this should be ready for a final look.

**Comment by mattmoor** [comment]
@damienmg If this LGTY then I'll merge things and get started on an import into the mono-repo.  I've retitled to be clear that this should be in a state I expect to merge :)

**Pull request merged**